### PR TITLE
Adding suport for -var-file and -target argument

### PR DESCRIPTION
**Pull request by aclowkey**
#192 #191

### CODE REVIEW

**Comment by aclowkey** [comment]
Any tips on how to make it simple as possible?

**Comment by aclowkey** [comment]
What do you think ? I didn't test yet, but Ill do as soon as I can.

File: modules/terraform/cmd.go [hunk 1]
```diff
@@ -17,6 +17,14 @@ func GetCommonOptions(options *Options, args ...string) (*Options, []string) {
 		args = append(args, "-no-color")
 	}
 
+	for _, varFile = range options.VarFiles {
```
Comment (brikis98) [inline modules/terraform/cmd.go]: Probably needs to be `varFile :=`. Same with `target` below.

File: test/terraform_basic_example_test.go [hunk 2]
```diff
@@ -13,6 +13,7 @@ func TestTerraformBasicExample(t *testing.T) {
 
 	expectedText := "foo"
 
+        // TODO create the terraform varfile from test
```
Comment (brikis98) [inline test/terraform_basic_example_test.go]: Looks like you already created a `varfile.tfvars`, so no need to create it!

**Comment by aclowkey** [comment]
I'm struggling with running the tests :S

For some reason go isn't aware of the change in the files..
```
ac@ac-PC:~/go/src/github.com/gruntwork-io/terratest/test$ go test -v -run TestTerraformBasicExample
# github.com/gruntwork-io/terratest/test
./terraform_basic_example_test.go:25:11: unknown field 'VarFiles' in struct literal of type terraform.Options
FAIL	github.com/gruntwork-io/terratest/test [build failed]
```

**Comment by brikis98** [comment]
That's weird. Did you set up your `GOPATH` correctly?

**Comment by brikis98** [comment]
Are all the tests added and passing now?

**Comment by aclowkey** [comment]
Yes

File: modules/terraform/format.go [hunk 3]
```diff
@@ -19,6 +23,15 @@ func FormatTerraformVarsAsArgs(vars map[string]interface{}) []string {
 	return formatTerraformArgs(vars, "-var")
 }
 
+// FormatTerraformArgs will format multiple args with the arg name (e.g. arg-file a arg-file b)
+func FormatTerraformArgs(argName string, args []string) []string {
+	argsList := []string{}
+	for _, varFile := range args {
```
Comment (brikis98) [inline modules/terraform/format.go]: `argValue` is probably a better name here than `varFile`

File: modules/terraform/format.go [hunk 4]
```diff
@@ -19,6 +23,15 @@ func FormatTerraformVarsAsArgs(vars map[string]interface{}) []string {
 	return formatTerraformArgs(vars, "-var")
 }
 
+// FormatTerraformArgs will format multiple args with the arg name (e.g. arg-file a arg-file b)
```
Comment (brikis98) [inline modules/terraform/format.go]: A more realistic example might help explain what this method does. E.g., 

```
FormatTerraformArgs("-var-file", []string{"foo.tfvars", "bar.tfvars"}) returns "-var-file foo.tfvars -var-file bar.tfvars"
```

File: test/terraform_basic_example_test.go [hunk 5]
```diff
@@ -1,26 +1,35 @@
...
 
 		// Variables to pass to our Terraform code using -var options
 		Vars: map[string]interface{}{
 			"example": expectedText,
 		},
+		// Only create the example1 file
+		Targets: []string{"local_file.example"},
+
+		// Use the var files
+		VarFiles: []string{"varfile.tfvars"},
```
Comment (brikis98) [inline test/terraform_basic_example_test.go]: Hm, `terraform_basic_example_test.go` should be, well, a basic example intended to provide simple example code for Terratest newbies. I think `Targets` and `VarFiles` and the corresponding checks may be a little too complex for someone just starting out. Perhaps move this example + corresponding test code to a different example?

**Comment by brikis98** [comment]
@aclowkey What is the latest status with this. Have all comments been resolved?

**Comment by kurtmaile** [comment]
Will this be merged soon by any chance?

**Comment by brikis98** [comment]
@aclowkey Did you add all tests/examples you were planning to?

**Comment by aclowkey** [comment]
Yes

**Comment by brikis98** [comment]
https://github.com/gruntwork-io/terratest/releases/tag/v0.13.27

**Pull request merged**